src/attachments/schemas/relationship.schema.ts

```diff
@@ -0,0 +1,40 @@
+import { Prop, Schema, SchemaFactory } from "@nestjs/mongoose";
+import { ApiProperty } from "@nestjs/swagger";
+import { Document } from "mongoose";
+import { AttachmentRelationTargetType } from "../types/relationship-filter.enum";
+
+export type RelationshipDocument = AttachmentRelationshipClass & Document;
+
+@Schema({ _id: false })
+export class AttachmentRelationshipClass {
+  @ApiProperty({
+    type: [String],
+    description: "Array of entity target IDs.",
+    default: [],
+  })
+  @Prop({ type: [String], default: [] })
+  targetIds: string[];
```

Comment:
I will suggest one target one relationship

Reply:
Could you give me a scenario where the suggested change would be beneficial? I'm assuming you are suggesting to target: {ids:[], type:""}
My stance here is, fewer nested fields are easier to understand and less code is needed

Reply:
Apologies for not being clear.
One entry in the relationships can describe only a relationship between one attachment and one dataset or one sample or one instrument.
If the attachment is attached to multiple datasets, you should have as many relationships as the number of datasets. One per datasets.
Let me know if I was able to explain it better